### DISCUSSION TITLE
[Question]: Preserving logs after Frigate restarts

### DISCUSSION THREAD

**Discussion by bluezclouds** [blakeblackshear/frigate#13150] [Ask A Question] [answered]
### What is your question:

My Frigate installation is crashing and restarting every 36 hours or so and I'm trying to figure out what is happening, but the logs are cleared when Frigate restarts.  Are older logs stored somewhere, or is there a setting that will preserve them between restarts?

Thanks.

**Comment by NickM-27** [comment]
you can use docker to map /dev/shm/logs to storage to persist them

**Reply by bluezclouds** [reply]
Thanks for the response, it points me in the right direction.  Now to figure out exactly how to do that.  It is amazing how complex Docker is.

**Reply by hawkeye217** [reply]
In the `volumes` section of your docker compose file, just add:

```
    volumes:
      ...
      - /path/to/any/directory:/dev/shm/logs
      ...
```

**Reply by bluezclouds** [reply]
I've searched the entire drive and there is only one docker-compose.yml file.  It's not in a Frigate container directory, it's in a /var/lib/docker/overlay2/ subdirectory.  

Is this perhaps different for Frigate containers installed as a Home Assistant Add-on?

**Reply by NickM-27** [reply, marked as answer]
If you're running as an addon then there is no way to do this

**Reply by bluezclouds** [reply]
That complicates things...  Are there any guides or methods to troubleshooting crash/restart issues with Frigate installed as an Add-On?

**Reply by NickM-27** [reply]
only using the frigate logs that are shown in the HA OS UI

**Reply by DawalaDeny** [reply]
> In the `volumes` section of your docker compose file, just add:
> 
> ```
>     volumes:
>       ...
>       - /path/to/any/directory:/dev/shm/logs
>       ...
> ```

Is there a way to make those logs persistent? I get the logs on my host pc, but when the container restarts, those logs are overridden by new ones.

**Reply by NickM-27** [reply]
They are persistent, the file rotates every 10MB

Docker also contains logs and you can view them all using `docker logs frigate -t`
